### PR TITLE
Before and after handler

### PR DESCRIPTION
**Pull request by malaupa**
- Introduces before and after handlers, which can be configured by the module configuration and which handle something before and after a request.
- Adds an enum with all operation names of a service.
- fixes dispatch errors, which now enables control if an error occurs for them

### CODE REVIEW

File: dist/index.es.js [hunk 1]
```diff
@@ -239,4 +240,4 @@ var TypedFormGroup = /** @class */ (function (_super) {
     return TypedFormGroup;
 }(FormGroup));
 
-export { BaseFormControlFactory, TypedFormControl, TypedFormGroup };
+export { BaseFormControlFactory, TypedFormGroup, TypedFormControl };
```
Comment (cogidoo) [inline dist/index.es.js]: Did you ˋnpm installˋ before building this packages?! Looks like you do not have the latest version of rollup?!

File: src/mustache/api.service.mustache [hunk 2]
```diff
@@ -347,7 +348,17 @@ export class {{classname}} {
...
 
-        if(typeof this.configuration.errorHandler === 'function') {
+        if (typeof this.configuration.beforeHandler === 'function') {
+          this.configuration.beforeHandler('{{operationIdOriginal}}', '{{httpMethod}}'.toUpperCase());
+        }
+        if (typeof this.configuration.afterHandler === 'function') {
+          handle = handle.pipe(tap(
+            result => this.configuration.afterHandler('{{operationIdOriginal}}', '{{httpMethod}}'.toUpperCase(), result),
+            err => this.configuration.afterHandler('{{operationIdOriginal}}', '{{httpMethod}}'.toUpperCase(), err)
+          ));
```
Comment (cogidoo) [inline src/mustache/api.service.mustache]: How can you (the code that uses this api-service) distinguish if the after-handler was called with result- or error-data? Wouldn´t it be better to call the after-handler with to parameters - result and error?

Comment (malaupa) [reply src/mustache/api.service.mustache]: It is not possible to call the after handler with two parameters. The result and error exists not at the same time. You can identify the error case by checking the type with `instanceof HttpErrorResponse`.

File: dist/index.es.js (hunk 1 above)
Comment (malaupa) [reply dist/index.es.js]: Done!

**Pull request merged**